.travis.yml

```diff
@@ -36,10 +35,10 @@ before_install:
   - python --version
   - doxygen --version
 install:
-  - pip install -r requirements.txt
-  - pip install pytest
-  - pip install pylint
-  - pip install hypothesis
-  - pip install mock
-  - pip install coverage
-  - pip install coveralls
+  - pip install --user -r requirements.txt
+  - pip install --user pytest
+  - pip install --user pylint
+  - pip install --user hypothesis
+  - pip install --user mock
+  - pip install --user coverage
+  - pip install --user coveralls
```

Comment:
Should not we have a requirements-travis.txt or requirements-test.txt file for theses? If not, why don't we install all packages in one pip install command?

Reply:
I don't think this list is big enough to need another level of indirection IMO.

I can run these all in one command though.

Reply:
Ok, for some reason combining all of the requirements into one pip install command caused travis to fail. I couldn't figure out why.

Reply:
If I am right, the error is `from six import wraps - ImportError: cannot import name wraps`.
[Here](https://stackoverflow.com/a/36073446/3272420) I found that Mock requires six version 1.7 or newer.
I think you should add `pip install --upgrade six` to upgrade the already installed six to the newest version.

Anyway, I'm being picky, if it's good enough for mbed's maintainers, let's merge as it is :)

Reply:
Huh, then why would pip work in this configuration?

Reply:
hum yeah, you are right ... duno ¯\(°_o)/¯